### PR TITLE
Trim whitespace between values when reading from configuration from runsettings

### LINKED ISSUE #641: In coverlet.collector, allow repeating configuration elements in runsettings
When excluding multiple elements, the current format can make things a bit unwieldy (and makes the changes much harder to track
```
<Exclude>[lib]ns1.*,[lib]ns2.*</Exclude>
```
It would be nice if we could specify multiple items on different Exclude nodes:
```
<Exclude>[lib]ns1.*</Exclude>
<Exclude>[lib]ns2.*</Exclude>
```
Same goes for the other colon separated settings

### PR DESCRIPTION
**Pull request by EricStG**
Closes #641

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
Thank's I'll review asap!

File: src/coverlet.collector/DataCollection/CoverletSettingsParser.cs [hunk 1]
```diff
@@ -205,5 +205,15 @@ private bool ParseIncludeTestAssembly(XmlElement configurationElement)
...
         }
+
+        /// <summary>
+        /// Splits a comma separated elements into an array
+        /// </summary>
+        /// <param name="element">The element to split</param>
+        /// <returns>An array of the values in the element</returns>
+        private string[] SplitElement(XmlElement element)
+        {
+            return element?.InnerText?.Split(',')?.Select(value => value.Trim())?.ToArray();
```
Comment (MarcoRossignoli) [inline src/coverlet.collector/DataCollection/CoverletSettingsParser.cs]: Maybe `return element?.InnerText?.Split(',', StringSplitOptions.RemoveEmptyEntries).Select(value => value.Trim()).ToArray();` why the null check after `InnerText` one?

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs [hunk 2]
```diff
@@ -42,17 +42,24 @@ public void ParseShouldSelectFirstTestModuleFromTestModulesList()
             Assert.Equal("module1.dll", coverletSettings.TestModule);
         }
 
-        [Fact]
-        public void ParseShouldCorrectlyParseConfigurationElement()
+        [Theory]
+        [InlineData("[*]*,[coverlet]*", "[coverlet.*.tests?]*,[coverlet.*.tests.*]*", @"E:\temp,/var/tmp", "module1.cs,module2.cs", "Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute")]
+        [InlineData("[*]*, [coverlet]*", "[coverlet.*.tests?]*, [coverlet.*.tests.*]*", @"E:\temp, /var/tmp", "module1.cs, module2.cs", "Obsolete, GeneratedCodeAttribute, CompilerGeneratedAttribute")]
+        [InlineData("[*]*,\r\n[coverlet]*", "[coverlet.*.tests?]*,\r\n[coverlet.*.tests.*]*", "E:\\temp,\r\n/var/tmp", "module1.cs,\r\nmodule2.cs", "Obsolete,\r\nGeneratedCodeAttribute,\r\nCompilerGeneratedAttribute")]
+        public void ParseShouldCorrectlyParseConfigurationElement(string includeFilters,
```
Comment (MarcoRossignoli) [inline test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Can you add some test with empty space and null?

File: src/coverlet.collector/DataCollection/CoverletSettingsParser.cs (hunk 1 above)
Comment (EricStG) [reply src/coverlet.collector/DataCollection/CoverletSettingsParser.cs]: I've updated it and added a coalesce operator at the end to return an empty array instead of null. The extra null checks weren't needed so they are gone.

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 2 above)
Comment (EricStG) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Added a few cases for missing elements, null inner texts and whitespace

File: src/coverlet.collector/DataCollection/CoverletSettingsParser.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/coverlet.collector/DataCollection/CoverletSettingsParser.cs]: > I've updated it and added a coalesce operator at the end to return an empty array instead of null. The extra null checks weren't needed so they are gone.

I prefer to keep old semantic and return null, I'd like to avoid surprise in case of expected null of callers. Please remove the coalesce at the end, the rest is ok to me.

Comment (MarcoRossignoli) [reply src/coverlet.collector/DataCollection/CoverletSettingsParser.cs]: It's also a shame, maybe in future we'll update with elegant one https://github.com/dotnet/corefx/issues/41487#issuecomment-573576875

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: thank's!

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs [hunk 3]
```diff
@@ -42,17 +42,27 @@ public void ParseShouldSelectFirstTestModuleFromTestModulesList()
...
         }
 
-        [Fact]
-        public void ParseShouldCorrectlyParseConfigurationElement()
+        [Theory]
+        [InlineData("[*]*,[coverlet]*", "[coverlet.*.tests?]*,[coverlet.*.tests.*]*", @"E:\temp,/var/tmp", "module1.cs,module2.cs", "Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute")]
+        [InlineData("[*]*,,[coverlet]*", "[coverlet.*.tests?]*,,[coverlet.*.tests.*]*", @"E:\temp,,/var/tmp", "module1.cs,,module2.cs", "Obsolete,,GeneratedCodeAttribute,,CompilerGeneratedAttribute")]
+        [InlineData("[*]*, ,[coverlet]*", "[coverlet.*.tests?]*, ,[coverlet.*.tests.*]*", @"E:\temp, ,/var/tmp", "module1.cs, ,module2.cs", "Obsolete, ,GeneratedCodeAttribute, ,CompilerGeneratedAttribute")]
+        [InlineData("[*]*, [coverlet]*", "[coverlet.*.tests?]*, [coverlet.*.tests.*]*", @"E:\temp, /var/tmp", "module1.cs, module2.cs", "Obsolete, GeneratedCodeAttribute, CompilerGeneratedAttribute")]
+        [InlineData("[*]*,\r\n[coverlet]*", "[coverlet.*.tests?]*,\r\n[coverlet.*.tests.*]*", "E:\\temp,\r\n/var/tmp", "module1.cs,\r\nmodule2.cs", "Obsolete,\r\nGeneratedCodeAttribute,\r\nCompilerGeneratedAttribute")]
```
Comment (MarcoRossignoli) [inline test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Please add also some test with space before and after CRLF for instance

`"   [*]*,    \r\n[coverlet]*"   , "[coverlet.*.tests?]*   ,   \r\n   [coverlet.*.tests.*]*   "`

Comment (EricStG) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Added those as well as a few tabs for good measure

File: src/coverlet.collector/DataCollection/CoverletSettingsParser.cs (hunk 1 above)
Comment (EricStG) [reply src/coverlet.collector/DataCollection/CoverletSettingsParser.cs]: Coalesce has been removed

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 3 above)
Comment (MarcoRossignoli) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Good catch...test fails on tabs

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs [hunk 4]
```diff
@@ -62,22 +78,68 @@ public void ParseShouldCorrectlyParseConfigurationElement()
...
+            Assert.Null(coverletSettings.ExcludeSourceFiles);
+            Assert.Null(coverletSettings.ExcludeAttributes);
+            Assert.Single(coverletSettings.ExcludeFilters, "[coverlet.*]*");
+        }
+
         [Theory]
         [InlineData(" , json", 1, new[] { "json" })]
         [InlineData(" , json, ", 1, new[] { "json" })]
         [InlineData("json,cobertura", 2, new[] { "json", "cobertura" })]
+        [InlineData("json,\r\ncobertura", 2, new[] { "json", "cobertura" })]
```
Comment (MarcoRossignoli) [inline test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Please add some space and tabs also here

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 3 above)
Comment (EricStG) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: That was an error in the test. `@"\t"` is not the same as `"\t"`. It's been fixed.

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 4 above)
Comment (EricStG) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: Added a few cases with space, tabs and a mix of everything

File: test/coverlet.collector.tests/CoverletSettingsParserTests.cs (hunk 3 above)
Comment (MarcoRossignoli) [reply test/coverlet.collector.tests/CoverletSettingsParserTests.cs]: I see
